pkg/daemon/daemon.go

```diff
@@ -1368,16 +1368,6 @@ func (dn *Daemon) validateOnDiskState(currentConfig *mcfgv1.MachineConfig) error
 	}
 }
 
-// compareOSImageURL checks whether the current and desired
-// URL are the same.  This used to do more, but now the
-// only special casing is to support an empty desired URL
-// as meaning "keep current OS" which we probably don't need
-// anymore either.
```

Comment:
Yeah, let's try nuking this.  I think I wrote that comment and was probably too conservative at the time.